### ISSUE TITLE
For attack_id, cwe_id, capec_id, cpe_id, mitre_id, cve_id endpoints, return 404

### ISSUE THREAD

**Issue by himynamesdave** [muchdogesec/vulmatch#39]
All endpoints that accept these IDs should return 404s if the object does not exist.

Currently return empty responses

**Comment by thekickasssperm** [comment]
Is this the same issue ? Experiencing the same on docker build

 ERROR: HTTP error 404 while getting https://github.com/muchdogesec/arango_cti_processor/archive/embedded-relationship-tests.zip#39
6.998 ERROR: Could not install requirement arango_cti_processor@ https://github.com/muchdogesec/arango_cti_processor/archive/embedded-relationship-tests.zip#39 from https://github.com/muchdogesec/arango_cti_processor/archive/embedded-relationship-tests.zip#39 (from -r requirements.txt (line 53)) because of HTTP error 404 Client Error: Not Found for url: https://codeload.github.com/muchdogesec/arango_cti_processor/zip/embedded-relationship-tests for URL https://github.com/muchdogesec/arango_cti_processor/archive/embedded-relationship-tests.zip#39

**Comment by himynamesdave** [comment]
> Is this the same issue ? Experiencing the same on docker build
> 
> ERROR: HTTP error 404 while getting https://github.com/muchdogesec/arango_cti_processor/archive/embedded-relationship-tests.zip#39 6.998 ERROR: Could not install requirement arango_cti_processor@ https://github.com/muchdogesec/arango_cti_processor/archive/embedded-relationship-tests.zip#39 from https://github.com/muchdogesec/arango_cti_processor/archive/embedded-relationship-tests.zip#39 (from -r requirements.txt (line 53)) because of HTTP error 404 Client Error: Not Found for url: https://codeload.github.com/muchdogesec/arango_cti_processor/zip/embedded-relationship-tests for URL https://github.com/muchdogesec/arango_cti_processor/archive/embedded-relationship-tests.zip#39

@thekickasssperm nope, its not. Can you try pulling the main branch and rebuilding? I think that should fix your issue. If you're still running into issues please head over to our community forum (https://community.dogesec.com/) and I'd be happy to help you get it running.

**Comment by thekickasssperm** [comment]
Thanks for the quick response:

Still running into the same issue 

>  => [celery internal] load build definition from Dockerfile                                                                                                                                                                 0.0s
 => => transferring dockerfile: 196B                                                                                                                                                                                        0.0s
 => [env_django internal] load build definition from Dockerfile                                                                                                                                                             0.0s
 => => transferring dockerfile: 196B                                                                                                                                                                                        0.0s
 => [celery internal] load metadata for docker.io/library/python:3.11                                                                                                                                                       0.8s
 => [celery internal] load .dockerignore                                                                                                                                                                                    0.0s
 => => transferring context: 2B                                                                                                                                                                                             0.0s
 => [env_django internal] load .dockerignore                                                                                                                                                                                0.0s
 => => transferring context: 2B                                                                                                                                                                                             0.0s
 => [env_django 1/4] FROM docker.io/library/python:3.11@sha256:3cdce69fd5663ca47c420ec4d4df8e3545519a4030372f7d2064fb1be2279844                                                                                             0.0s
 => [celery internal] load build context                                                                                                                                                                                    0.0s
 => => transferring context: 38B                                                                                                                                                                                            0.0s
 => [env_django internal] load build context                                                                                                                                                                                0.0s
 => => transferring context: 38B                                                                                                                                                                                            0.0s
 => CACHED [env_django 2/4] WORKDIR /usr/src/app                                                                                                                                                                            0.0s
 => CACHED [env_django 3/4] COPY requirements.txt ./                                                                                                                                                                        0.0s
 => ERROR [celery 4/4] RUN pip install -r requirements.txt                                                                                                                                                                  3.8s
------                                                                                                                                                                                                                           
 > [celery 4/4] RUN pip install -r requirements.txt:                                                                                                                                                                             
0.740 Collecting stix2arango@ https://github.com/muchdogesec/stix2arango/archive/main.zip (from -r requirements.txt (line 45))                                                                                                   
1.396   Downloading https://github.com/muchdogesec/stix2arango/archive/main.zip                                                                                                                                                  
2.582      \ 5.5 MB 4.7 MB/s 0:00:01                                                                                                                                                                                             
2.664   Installing build dependencies: started                                                                                                                                                                                   
3.388   Installing build dependencies: finished with status 'done'
3.389   Getting requirements to build wheel: started
3.419   Getting requirements to build wheel: finished with status 'done'
3.419   Preparing metadata (pyproject.toml): started
3.474   Preparing metadata (pyproject.toml): finished with status 'done'
3.476 Collecting arango_cti_processor@ https://github.com/muchdogesec/arango_cti_processor/archive/embedded-relationship-tests.zip#39 (from -r requirements.txt (line 53))
3.747   ERROR: HTTP error 404 while getting https://github.com/muchdogesec/arango_cti_processor/archive/embedded-relationship-tests.zip#39
3.748 ERROR: Could not install requirement arango_cti_processor@ https://github.com/muchdogesec/arango_cti_processor/archive/embedded-relationship-tests.zip#39 from https://github.com/muchdogesec/arango_cti_processor/archive/embedded-relationship-tests.zip#39 (from -r requirements.txt (line 53)) because of HTTP error 404 Client Error: Not Found for url: https://codeload.github.com/muchdogesec/arango_cti_processor/zip/embedded-relationship-tests for URL https://github.com/muchdogesec/arango_cti_processor/archive/embedded-relationship-tests.zip#39
3.759 
3.759 [notice] A new release of pip is available: 24.0 -> 24.2
3.759 [notice] To update, run: pip install --upgrade pip
------
failed to solve: process "/bin/sh -c pip install -r requirements.txt" did not complete successfully: exit code: 1

**Comment by fqrious** [comment]
You removed a branch that we depend on, changing it back to main will fix

**Comment by himynamesdave** [comment]
@fqrious I can see we pull in https://github.com/muchdogesec/arango_cti_processor/archive/embedded-relationship-tests.zip for ACTIP but where is this defined?

It should now be https://github.com/muchdogesec/arango_cti_processor/archive/main.zip

**Comment by fqrious** [comment]
requirements.txt

**Comment by thekickasssperm** [comment]
Do you mean the last line of requirements.txt?

**Comment by fqrious** [comment]
Yes

**Comment by himynamesdave** [comment]
this was a series of mess ups on my parts. It's fixed now

https://github.com/muchdogesec/vulmatch/commit/ace3876ab5f2f70a944da131b327bc2fc63c5ce8

**Comment by thekickasssperm** [comment]
Thanks for the quick assists! But I didn't have a successful attempt running it on Macos M3, is it supported on mac ?

**Comment by himynamesdave** [comment]
Please move these to our community forum @thekickasssperm  (https://community.dogesec.com/). I don't want to confuse the thread related to the issue this ticket covers
